### ISSUE TITLE
_FillValue and missing_value mismatch if float

### ISSUE THREAD

**Issue by pacioos** [pydata/xarray#1601]
I get the following error when _FillValue and missing_value are both floating-point decimals, even though they are the same value (-999.99):

```
  File "/usr/local/lib/python2.7/dist-packages/xarray/backends/api.py", line 301, in open_dataset
    return maybe_decode_store(store, lock)
  File "/usr/local/lib/python2.7/dist-packages/xarray/backends/api.py", line 225, in maybe_decode_store
    drop_variables=drop_variables)
  File "/usr/local/lib/python2.7/dist-packages/xarray/conventions.py", line 955, in decode_cf
    decode_coords, drop_variables=drop_variables)
  File "/usr/local/lib/python2.7/dist-packages/xarray/conventions.py", line 888, in decode_cf_variables
    decode_times=decode_times)
  File "/usr/local/lib/python2.7/dist-packages/xarray/conventions.py", line 800, in decode_cf_variable
    "and decoding explicitly using "
ValueError: ('Discovered conflicting _FillValue and missing_value.  Considering opening the offending dataset using decode_cf=False, corrected the attributes', 'and decoding explicitly using xarray.conventions.decode_cf(ds)')
```

Here is the OPeNDAP end point that causes the error:

http://oos.soest.hawaii.edu/thredds/dodsC/hioos/tide_mhi

The error is unwarranted since the values match. My workaround thus far has been to first open the dataset with decode_cf=False, then I remove the duplicate missing_value attributes, and then decode explicitly using decode_cf(ds).

**Comment by shoyer** [comment]
Inspecting your dataset, I see:
```
In [7]: ds = xarray.open_dataset('http://oos.soest.hawaii.edu/thredds/dodsC/hioos/tide_mhi', decode_cf=False)

In [8]: ds.info()
xarray.Dataset {
dimensions:
	depth = 3 ;
	latitude = 242 ;
	longitude = 521 ;
	time = 8760 ;

variables:
	...
	float64 v(time, depth, latitude, longitude) ;
		v:units = meter second-1 ;
		v:long_name = v-velocity component ;
		v:standard_name = northward_sea_water_velocity ;
		v:missing_value = -999.989990234 ;
		v:_FillValue = -999.99 ;
	...
```

So `missing_value` and `_FillValue` do indeed differ here, at least in their 64-bit floating point representation. Floating point equality is a pain, but unfortunately for fill values we really do need the bit-patterns to match.

https://github.com/pydata/xarray/pull/1602 updates the error message to print the offending values, which should hopefully improve user-experience if this comes up in the future.

**Comment by shoyer** [comment]
Fixed by #1602
